pyvista/plotting/colors.py

```diff
@@ -348,10 +377,473 @@
+        >>> c
+        Color(hex='#ff0000ff')
+        >>> c.f_rgb
+        (1.0, 0.0, 0.0)
+
+        """
+        return self.f_rgba[:3]
+
+    @property
+    def hex(self) -> str:
+        """Get the color value as an RGBA hexadecimal value.
+
+        Examples
+        --------
+        Create a blue color with half opacity.
+
+        >>> import pyvista
+        >>> c = pyvista.Color("blue", default_opacity="#80")
+        >>> c
+        Color(name='blue', hex='#0000ff80')
+
+        Create a transparent red color using an RGBA hexadecimal value.
+
+        >>> c = pyvista.Color("0xff000040")
+        >>> c
+        Color(hex='#ff000040')
+
+        """
+        return '#' + ''.join(
+            f"{c:0>2x}" for c in (self._red, self._green, self._blue, self._opacity)
+        )
+
+    @property
+    def name(self) -> Optional[str]:
+        """Get the color name.
+
+        Notes
+        -----
+        Refer to the table below for a list of supported colors.
+
+        .. include:: ../colors.rst
+
+        Examples
+        --------
+        Create a blue color with half opacity.
+
+        >>> import pyvista
+        >>> c = pyvista.Color("blue", default_opacity=0.5)
+        >>> c
+        Color(name='blue', hex='#0000ff80')
+
+        """
+        return self._name
+
+    @property
+    def vtk_c3ub(self) -> _vtk.vtkColor3ub:
+        """Get the color value as a VTK Color3ub instance.
+
+        Examples
+        --------
+        Create a blue color with half opacity.
+
+        >>> import pyvista
+        >>> c = pyvista.Color("blue", default_opacity=0.5)
+        >>> c
+        Color(name='blue', hex='#0000ff80')
+        >>> c.vtk_c3ub
+        vtkmodules.vtkCommonDataModel.vtkColor3ub([0, 0, 255])
+
+        """
+        return _vtk.vtkColor3ub(self._red, self._green, self._blue)
+
+    def linear_to_srgb(self):
+        """Convert from linear color values to sRGB color values.
+
+        Returns
+        -------
+        Color
+            A new Color instance with sRGB color values.
+
+        """
+        rgba = np.array(self.f_rgba)
+        mask = rgba < 0.0031308
+        rgba[mask] *= 12.92
+        rgba[~mask] = 1.055 * rgba[~mask] ** (1 / 2.4) - 0.055
+        return Color(rgba)
+
+    def srgb_to_linear(self):
+        """Convert from sRGB color values to linear color values.
+
+        Returns
+        -------
+        Color
+            A new Color instance with linear color values.
+
+        """
+        rgba = np.array(self.f_rgba)
+        mask = rgba < 0.04045
+        rgba[mask] /= 12.92
+        rgba[~mask] = ((rgba[~mask] + 0.055) / 1.055) ** 2.4
+        return Color(rgba)
+
+    @classmethod
+    def from_dict(cls, dict_):
+        """Construct from dictionary for JSON deserialization."""
+        return Color(dict_)
+
+    def to_dict(self):
+        """Convert to dictionary for JSON serialization."""
+        return (self._red, self._green, self._blue, self._opacity)
```

Comment:
This is not a dict.

Reply:
I know, but the only reason these are added is to keep the theme's serializable to/from JSON:

https://github.com/pyvista/pyvista/blob/2df0690cea486a60dd747b8d7c477f15e61e2425/pyvista/themes.py#L2154-L2174

And it works just fine for that purpose. I can make it a proper dict if you prefer though.
What I use in some personal projects is a pair of `to_cfg` and `from_cfg` methods and then the serializable subclasses can determine themselves whether they should return a dict, tuple, str, int or float to serialize the instance.

Reply:
That's a bit of a pickle :P I really wouldn't like to have a `to_dict` method that returns a tuple. If serialisation is always through `themes.py` we could amend the Themes' `to_dict()` at https://github.com/pyvista/pyvista/blob/2df0690cea486a60dd747b8d7c477f15e61e2425/pyvista/themes.py#L173-L176

Add a case where the value has a `to_tuple()` or whatever method we _do_ want to define on our `Color`, and use that in a new branch in that `if`. Does this sound reasonable?

Reply:
Let's keep things simple, I will just return a proper dict and construct a `Color` from such a dict :)